src/acceptor.rs

```diff
@@ -0,0 +1,676 @@
+    ///   - accepted is NULL.
+    ///   - rustls_accepted_into_connection has already been called with `accepted`.
+    #[no_mangle]
+    pub extern "C" fn rustls_accepted_signature_scheme(
+        accepted: *const rustls_accepted,
+        i: usize,
+    ) -> u16 {
+        ffi_panic_boundary! {
+            let accepted: &Option<Accepted> = try_ref_from_ptr!(accepted);
+            let accepted = match accepted {
+                Some(a) => a,
+                None => return 0,
+            };
+            let hello = accepted.client_hello();
+            let signature_schemes = hello.signature_schemes();
+            match signature_schemes.get(i) {
+                Some(s) => s.get_u16(),
+                None => 0,
+            }
+        }
+    }
+
+    /// Get the i'th in the list of cipher suites offered in the ClientHello.
+    ///
+    /// Parameters:
+    ///
+    /// accepted: The rustls_accepted to access.
+    /// i: Fetch the cipher suite at this offset.
+    ///
+    /// Returns:
+    ///
+    /// A cipher suite value from <https://www.iana.org/assignments/tls-parameters/tls-parameters.xhtml#tls-parameters-4.>
+    ///
+    /// This will be 0 in these cases:
+    ///   - i is greater than the number of available cipher suites.
+    ///   - accepted is NULL.
+    ///   - rustls_accepted_into_connection has already been called with `accepted`.
+    ///
+    /// Note that 0 is technically a valid cipher suite "TLS_NULL_WITH_NULL_NULL",
+    /// but this library will never support null ciphers.
+    #[no_mangle]
+    pub extern "C" fn rustls_accepted_cipher_suite(
+        accepted: *const rustls_accepted,
+        i: usize,
+    ) -> u16 {
+        ffi_panic_boundary! {
+            let accepted: &Option<Accepted> = try_ref_from_ptr!(accepted);
+            let accepted = match accepted {
+                Some(a) => a,
+                None => return 0,
+            };
+            let hello = accepted.client_hello();
+            let cipher_suites = hello.cipher_suites();
+            match cipher_suites.get(i) {
+                Some(cs) => cs.get_u16(),
+                None => 0,
+            }
+        }
+    }
+
+    /// Get the i'th in the list of ALPN protocols requested in the ClientHello.
+    ///
+    /// accepted: The rustls_accepted to access.
+    /// i: Fetch the ALPN value at this offset.
+    ///
+    /// Returns:
+    ///
+    /// A rustls_slice_bytes containing the i'th ALPN protocol. This may
+    /// contain internal NUL bytes and is not guaranteed to contain valid
+    /// UTF-8.
+    ///
+    /// This will be a zero-length rustls_slice bytes in these cases:
+    ///   - i is greater than the number of offered ALPN protocols.
+    ///   - The client did not offer the ALPN extension.
+    ///   - The `accepted` parameter was already transformed into a connection
+    ///      with rustls_accepted_into_connection.
+    ///   
+    /// The returned value is valid until the next time a method is called
```

Comment:
Maybe make it explicit that the `'static` lifetime on the return value is thus fake and must not be relied upon?